File: framework/src/main/java/org/example/loader/IntentDispatcher.java
package org.example.loader;

import org.example.core.intent.IIntent;
import org.example.core.dispatcher.IIntentDispatcher;
import org.example.core.dispatcher.IPageDispatcher;

class IntentDispatcher implements IIntentDispatcher {
    private final IPageDispatcher dispatcher;

    IntentDispatcher(IPageDispatcher dispatcher) {
        this.dispatcher = dispatcher;
    }

    @Override
    public void dispatch(IIntent intent) {
        dispatcher.dispatch(intent);
    }
}

File: framework/src/main/kotlin/org/example/loader/IntentDispatcher.kt
package org.example.loader

import org.example.core.dispatcher.IIntentDispatcher
import org.example.core.dispatcher.IPageDispatcher
import org.example.core.intent.IIntent

internal class IntentDispatcher(private val dispatcher: IPageDispatcher) : IIntentDispatcher {

    override fun dispatch(intent: IIntent?) {
        dispatcher.dispatch(intent)
    }
}
